Added SNP data and SNP subjects export.

gijskant opened the pull request:
The actual exporters are in transmartApp. This commit contains
the enabling data querying changes.

cataphract commented:
Besides the obvious problems -- no tests, code duplication, misleading comments arising from copy-paste (have I mentioned the nonexistence of tests) and some less obvious ones -- the creation of a new data type resource to represent the **same** data type as snp_lz, the unacknowledged weakening of the post-conditions of an API call, a row type that doesn't implement `DataRow`  --, I still cannot understand what the purpose of snp_subjects is, besides looking like a complete subversion of the API.

Can you explain what you're trying to accomplish with this changeset? There's no explanation to be found either in the commit message or in comments in the code.

gijskant commented on grails-app/domain/org/transmartproject/db/dataquery/highdim/snp_lz/DeSnpInfo.groovy:
```diff
@@ -4,7 +4,7 @@ class DeSnpInfo {
 
     String name
     String chromosome
-    BigDecimal pos
```
BigDecimal provides "immutable, arbitrary-precision signed decimal numbers". A SNP position is an integer (Number of size 10 in the database design of DEAPP.DE_SNP_INFO, column CHROM_POS).

gijskant commented on src/groovy/org/transmartproject/db/dataquery/highdim/HighDimensionResourceService.groovy:
```diff
@@ -89,13 +89,11 @@ class HighDimensionResourceService implements HighDimensionResource {
 
         HashMultimap multiMap = HashMultimap.create()
```
These changes enable multiple modules to be applicable for the same 'platform'. I.e., these modules query different datatypes (subject data, SNPs), but these datatypes may be related to the same platform, e.g., SNP data.

gijskant commented on src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzModule.groovy:
```diff
@@ -171,7 +175,11 @@ class SnpLzModule extends AbstractHighDimensionDataTypeModule {
                 property 'snp.trialName',              'trialName'
 
```
It looks like the DeSnpInfo table contains that info related to SNPs. Am I mistaken?

cataphract replied:
The geneInfo? It's possible, but in the documentation the client gave us they describe snp_id (the one that links to `de_snp_info`) as a legacy column, so I avoided using it.

gijskant committed:
Prevent querying for patients one by one when fetching assays.

gijskant committed:
Enable SNP export, fix bugs in SNP data types.

- Add chromosome and chromosome position fields to SnpLzModule;
- Fix chromosome position in DeSnpInfo to be of type Integer;
- Fix SnpLzRow to iterator over SnpLzCell elements.

gijskant committed:
Updated integration tests for SnpLzModule.
